compiler/damlc/daml-package-config/src/DA/Daml/Package/Config.hs

```diff
@@ -66,6 +68,16 @@ parseProjectConfig project = do
     pSdkVersion <- queryProjectConfigRequired ["sdk-version"] project
     Right PackageConfigFields {..}
 
+checkPkgConfig :: PackageConfigFields -> [String]
+checkPkgConfig PackageConfigFields {pName, pVersion} =
+  [ "WARNING: Package names should have the format ^[a-zA-Z]([a-zA-Z\\-])*$. You may be able to compile packages with different formats, but you will not be able to use them as dependencies in other projects. Unsupported package names may start causing compilation errors without warning."
+  | not $ LF.unPackageName pName =~ ("^[a-zA-Z]([a-zA-Z\\-])*$" :: T.Text)
```

Comment:
I don't have strong opinions on what the allowed package names should be, but I think it should be simple (easy to explain & document) and compatible with GHC/Cabal package names (a subset, maybe a strict subset).

I can't seem to find where the allowed package names for Cabal are documented...

Reply:
updated the regex to allow digits and disallow trailing dashes. not quiet sure yet how to express the other constraints.

Reply:
I think the strict subset captured by
```
[A-Za-z][A-Za-z0-9]*(\-[A-Za-z][A-Za-z0-9]*)*
```
It's a bit strict than `Cabal` because it doesn't allow a component to be named `5x` but I don't think that's a problem.

Reply:
Changed it to the above regex.